lab_1/main.py

```diff
@@ -86,10 +165,40 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
-
-
-def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int) -> str or None:
+    stats = {}
+    try:
+        top_unknown = get_top_n_words(unknown_profile["freq"], top_n)
+        top_compare = get_top_n_words(profile_to_compare["freq"], top_n)
+        stats["name"] = profile_to_compare["name"]
+
+        cross = 0
+        stats["common"] = []
+        for token in top_compare:
+            if token in top_unknown:
+                cross += 1
+                stats["common"].append(token)
+
+        stats["score"] = round(cross / top_n, 2)
+
+        stats["sorted_common"] = sorted(stats["common"])
+        tokens_length = 0
+        max_l = len(list(profile_to_compare["freq"].keys())[0])
```

Comment:
explain this, please

Reply:
Firstly, I create an element with a key sorted_common by, evidently, sorting the list with most common words which I had already created earlier. 

Then, at tokens_length, I start caculating average length of the tokens for eventual dict which will be returned at the end. 

Here too I start looking for the longest word by assigning to max_l (short for length, of course) the first word in the list of keys, which was taken from the whole dictionary of frequencies of a language. 

I was confused where should I look for this word, however looking in unknown profile resulted in test errors and when I see what was expected as a correct result I realized I would have to look for it in profile_to_compare. There was no word needed for the longest in the unknown language. If that was also your question.